platform/ios/src/MGLMapView.mm

```diff
@@ -3550,8 +3550,12 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
         {
             id <MGLAnnotation> annotation = [self annotationWithTag:annotationTag];
             NSAssert(annotation, @"Unknown annotation found nearby tap");
```

Comment:
I think it is a bug that is an edge case 😄  
But, the fix would require a callback feature for at least some of the mbgl APIs right? The platforms should not actually update their state until _after_ core has completed updating its state (i.e. removing annotations so that queries of visible annotations won't include them)